### PR TITLE
feat: add reset option (cli/ui)

### LINKED ISSUE #950: rebuild failed
**What happened?**  
[14:09:10] info Workspace dev-con-test-gh already exists
[14:09:10] info Creating devcontainer...
[14:09:10] info fatal: destination path '/Users/jlovas/.devpod/agent/contexts/default/workspaces/dev-con-test-gh/content' already exists and is not an empty directory.
[14:09:10] error Cloning failed: clone repository: error cloning repository: exit status 128. Trying cloning on local machine and uploading folder
[14:09:10] info Cloning from host and upload folder to server
[14:09:18] info cloning local folder: decompress: create /Users/jlovas/.devpod/agent/contexts/default/workspaces/dev-con-test-gh/content/.git/objects/pack/pack-dd5fa7e931a655538029acc2ac4270d6939593ea.idx: open /Users/jlovas/.devpod/agent/contexts/default/workspaces/dev-con-test-gh/content/.git/objects/pack/pack-dd5fa7e931a655538029acc2ac4270d6939593ea.idx: permission denied
[14:09:18] error Try enabling Debug mode under Settings to see a more verbose output
[14:09:18] fatal run agent command: Process exited with status 1

**What did you expect to happen instead?**  
Rebuild my existing container.

**How can we reproduce the bug?** (as minimally and precisely as possible)  
Try rebuild an existing container.

My *`devcontainer.json`*:
```
{
  "dockerFile": "Dockerfile",
   "runArgs": ["--name", "my-bookworm-slim-ws"],
  "workspaceMount": "source=${localWorkspaceFolder},target=/workspace,type=bind",
  "workspaceFolder": "/workspace",
  "waitFor": "postCreateCommand",
  "customizations": {
    "codespaces": {
      "openFiles": []
    },
    "vscode": {
      "extensions": [
        "ms-azuretools.vscode-docker",
        "ms-toolsai.jupyter",
        "ms-toolsai.jupyter-keymap",
        "ms-toolsai.jupyter-renderers",
        "ms-python.black-formatter",
        "ms-python.python",
        "ms-python.debugpy",
        "ms-python.vscode-pylance",
        "kevinrose.vsc-python-indent",
        "matangover.mypy",
        "littlefoxteam.vscode-python-test-adapter"
      ]
    }
  },
}```

**Local Environment:**  
- DevPod Version: 0.5.4 (20240226.070850)
- Operating System: mac
- ARCH of the OS: AMD64 

**DevPod Provider:**  
- Local/remote provider: docker

**Anything else we need to know?**  
(I'm just testing dev container and DevPod features.)
The biggest problem with this is that my workspace will disappear.

### PR DESCRIPTION
**Pull request by 89luca89**
Add a CLI `--reset` flag, this will rebuild+wipe git repo and clone fresh sources.
`--recreate` flag will only re-build the container workspace, keeping sources in place.

image
image

Add GUI button to reset:

image

And make it clear uncommitted data will be lost:

image

Resolves ENG-2954
Fix #950

### CODE REVIEW

**Comment by 89luca89** [comment]
Also @pascalbreuninger I've added this "recycle" icon for the reset, let me know if this is ok I couldn't think of anything better :shrug:

**Comment by pascalbreuninger** [comment]
@89luca89  what do you think of 
```
<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" strokeWidth={1.5} stroke="currentColor" className="w-6 h-6">
  <path strokeLinecap="round" strokeLinejoin="round" d="M21 16.811c0 .864-.933 1.406-1.683.977l-7.108-4.061a1.125 1.125 0 0 1 0-1.954l7.108-4.061A1.125 1.125 0 0 1 21 8.689v8.122ZM11.25 16.811c0 .864-.933 1.406-1.683.977l-7.108-4.061a1.125 1.125 0 0 1 0-1.954l7.108-4.061a1.125 1.125 0 0 1 1.683.977v8.122Z" />
</svg>
```
as the icon?

**Comment by 89luca89** [comment]
Oh like a rewind, cool for me, also what do you think of a simple "undo" icon? (like the jump-back arrow in some editors) just asking I'm not that good at design :sweat_smile:

File: pkg/devcontainer/run.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ type UpOptions struct {
 func (r *runner) Up(ctx context.Context, options UpOptions) (*config.Result, error) {
 	// download workspace source before recreating container
 	_, isDockerDriver := r.Driver.(driver.DockerDriver)
-	if options.Recreate && !isDockerDriver {
+	if (options.Recreate || options.Reset) && !isDockerDriver {
```
Comment (pascalbreuninger) [inline pkg/devcontainer/run.go:94]: shouldn't `options.Recreate` already be true because we set it on the command earlier already

Comment (89luca89) [reply pkg/devcontainer/run.go:94]: it was a "just in case" thing but yea it should

File: desktop/yarn.lock
Comment (pascalbreuninger) [inline desktop/yarn.lock:1]: could you remove this diff please, looks like it's a leftover from installing with npm

**Comment by pascalbreuninger** [comment]
> Oh like a rewind, cool for me, also what do you think of a simple "undo" icon? (like the jump-back arrow in some editors) just asking I'm not that good at design 😅

also okay, but please make sure you're taking it from https://heroicons.com/ to keep it consistent with existing icons

File: desktop/yarn.lock
Comment (89luca89) [reply desktop/yarn.lock:1]: done

File: pkg/devcontainer/run.go (hunk 1 above)
Comment (89luca89) [reply pkg/devcontainer/run.go]: fixed

**Review by pascalbreuninger — APPROVED** [review]

**Review by pascalbreuninger — APPROVED** [review]

**Pull request merged**